contracts/test/integration/L2PlusSubnet.t.sol

```diff
@@ -0,0 +1,857 @@
+            recipientAddr: 0x53c82507aA03B1a6e695000c302674ef1ecb880B,
+            amount: 3,
+            expectedAmount: 0,
+            expectedOutcome: OutcomeType.ActorErr,
+            expectedRet: abi.encodeWithSelector(InvalidSubnetActor.selector)
+        });
+
+        sendCrossMessageFromParentToChildWithResult(params);
+    }
+
+    function testL2PlusSubnet_Token_SendCrossMessageFromParentToNonExistingActorError() public {
+        MockIpcContractResult caller = new MockIpcContractResult();
+        Params memory params = Params({
+            root: rootNetwork,
+            subnet: tokenSubnet,
+            subnetL3: nativeL3SubnetsWithTokenParent[0],
+            caller: caller,
+            callerAddr: address(caller),
+            recipientAddr: 0x53c82507aA03B1a6e695000c302674ef1ecb880B,
+            amount: 3,
+            expectedAmount: 0,
+            expectedOutcome: OutcomeType.ActorErr,
+            expectedRet: abi.encodeWithSelector(InvalidSubnetActor.selector)
+        });
+
+        sendCrossMessageFromParentToChildWithResult(params);
+    }
+
+    function fundSubnet(
+        GatewayDiamond gateway,
+        TestSubnetDefinition memory subnet,
+        address callerAddr,
+        uint256 amount
+    ) internal {
+        Asset memory subnetSupply = subnet.subnetActor.getter().supplySource();
+        if (subnetSupply.kind == AssetKind.ERC20) {
+            IERC20(subnetSupply.tokenAddress).approve(address(gateway), amount);
+            gateway.manager().fundWithToken(subnet.id, FvmAddressHelper.from(callerAddr), amount);
+        } else {
+            gateway.manager().fund{value: amount}(subnet.id, FvmAddressHelper.from(callerAddr));
+        }
+    }
+
+    function registerSubnet(address subnetActorAddr, GatewayDiamond gateway) internal {
+        vm.deal(subnetActorAddr, DEFAULT_COLLATERAL_AMOUNT);
+        vm.prank(subnetActorAddr);
+        registerSubnetGW(DEFAULT_COLLATERAL_AMOUNT, subnetActorAddr, gateway);
+        vm.stopPrank();
+    }
+
+    function sendCrossMessageFromChildToParentWithResult(Params memory params) public {
+        // register L2 into root nerwork
+        registerSubnet(params.subnet.subnetActorAddr, params.root.gateway);
+        // register L3 into L2 subnet
+        registerSubnet(params.subnetL3.subnetActorAddr, params.subnet.gateway);
+
+        vm.deal(params.callerAddr, 1 ether);
+        vm.prank(params.callerAddr);
+
+        fundSubnet(params.root.gateway, params.subnet, params.callerAddr, 100000);
+        fundSubnet(params.subnet.gateway, params.subnetL3, params.callerAddr, 100000);
+
+        // create the xnet message on the subnet L3 - it's local gateway
+        IpcEnvelope memory crossMessage = TestUtils.newXnetCallMsg(
+            IPCAddress({subnetId: params.subnetL3.id, rawAddress: FvmAddressHelper.from(params.callerAddr)}),
+            IPCAddress({subnetId: params.root.id, rawAddress: FvmAddressHelper.from(params.recipientAddr)}),
+            params.amount,
```

Comment:
the param amount could exceed the `fundSubnet` amount?

Reply:
Good catch!